lib/plugins/output-filter/kubernetes-enrichment.js

```diff
@@ -0,0 +1,212 @@
+    data.stLogEnabled = false
+    return
+  }
+  var annotations = podCache[key].metadata.annotations
+  if (annotations) {
+    var logsEnabled = annotations['sematext.com/logs-enabled']
+    if (FALSE_REGEX.test(logsEnabled)) {
+      data.stLogEnabled = false
+      podCache[key].stLogEnabled = false
+      if (process.env.DEBUG) {
+        console.error('stLogEnabled = false', key)
+      }
+    } else {
+      podCache[key].stLogEnabled = true
+    }
+  }
+}
+
+function addLogsIndex (key, data) {
+  if (podCache[key].stLogsTokenSet === false) {
+    return
+  }
+  // get ST Logs Token from pod annotation
+  var index = podCache[key].metadata.annotations['sematext.com/logs-token']
+  // get comma separated list of fields from pod annotation
+  if (index) {
+    data._index = index
+  } else {
+    podCache[key].stLogsTokenSet = false
+  }
+}
+/*
+
+function updatePodUid (key, data) {
+  if (data.kubernetes && podCache[key]) {
+    data.kubernetes.pod.uid = podCache[key].metadata.uid
+  } else {
+    podCache[key].stLogsTokenSet = false
+  }
+}
+
+*/
+
+function replaceDockerImageName (key, data) {
+  if (data.kubernetes && data.container && podCache[key].stImageCache) {
+    var containerName = data.kubernetes.pod.container.name
+    var imageInfo = podCache[key].stImageCache[data.kubernetes.pod.name + containerName]
+    if (!imageInfo) {
+      return
+    }
+    if (data.container.image.name === 'sha256') {
+      data.container.image.digest = 'sha256:' + data.container.image.tag
+    }
+    data.container.image.name = imageInfo.name
+    data.container.image.tag = imageInfo.tag
+    // data.container.image.registry = imageInfo.version
+    // data.container.image.name = imageInfo.plainImageName
+  } else {
+    podCache[key].stLogsTokenSet = false
+  }
+}
+
+function processAnnotations (data, context) {
+  var key = data.kubernetes.namespace + '/' + data.kubernetes.pod.name
+  if (podCache[key] && podCache[key].metadata) {
+    checkLogsEnabled(key, data, context)
+    if (data.stLogEnabled === false) {
+      // logs will be droped, so no need for other processing
+      return
+    }
+    // updatePodUid (key, data)
+    replaceDockerImageName(key, data)
+    removeFields(key, data)
+    addLogsIndex(key, data)
+  }
+}
+
+function enrichLogs (context, config, eventEmitter, data, callback) {
+  if (data.kubernetes && data.kubernetes.pod) {
+    if (podCache[data.kubernetes.namespace + '/' + data.kubernetes.pod.name]) {
+      processAnnotations(data, context)
+      if (data.stLogEnabled === false) {
+        // allow input plugins to close the input stream, e.g. input/docker/docker.js
+        eventEmitter.emit('dropLogsRequest', context, data)
+        if (config.debug == true) { 
+          console.log('logs dropped ', data.kubernetes.namespace + '/' + data.kubernetes.pod.name, data.message)
+        }
+        return callback()
+      } else {
+        return callback(null, data)
+      }
+    } else {
+      getPodSpec(
```

Comment:
Check my comment on line `40`. This is where you could have: 
`const pod = await getPodSpec(...)`
Then use the `pod` for everything. You avoid using a `function` at all.

Reply:
see reply to `comment on line 40`. Sure we can try to refactor, maybe in a different PR? As I mention the calling function needs to be declared as async as well, and the LA output-filter functions are not designed to be declared as `async`, so I guess we would need a larger refactoring in the framework and potentially in plugins.

Reply:
Understood. Leaving it for a future refactor makes perfect sense.